src/librustc_parse/parser/ty.rs

```diff
@@ -461,6 +452,27 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parses a lifetime ("outlives") bound, e.g. `'a`, according to:
+    /// ```
+    /// LT_BOUND = LIFETIME
+    /// ```
+    fn parse_generic_lt_bound(
+        &mut self,
+        lo: Span,
+        inner_lo: Span,
+        has_parens: bool,
+        question: Option<Span>,
+    ) -> PResult<'a, GenericBound> {
+        self.error_opt_out_lifetime(question);
+        let bound = GenericBound::Outlives(self.expect_lifetime());
+        if has_parens {
+            // FIXME(Centril): Consider not erroring here and accepting `('lt)` instead,
```

Comment:
`('lt)` is entirely useless and it's some PITA to parse it in the first trait object position like `type A = `('lt) + OtherBounds`, so it isn't accepted anywhere.